### PR TITLE
[FLINK-28807] Honor schema lifecycle 

### PR DESCRIPTION
**Pull request by zentol**
This PR resolves a large number of violations of the schema lifecycle; where the schema was used without open() being called beforehand.
This is generally bad because certain schemas might rely on it, but also a blocker for FLINK-28621 because it will require the mapper to be created post-deserialization (where `open()` is the appropriate olace to do this).

### CODE REVIEW

File: flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testutils/formats/SchemaUtils.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.testutils.formats;
+
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+
+/** TODO: Add javadoc. */
```
Comment (alpinegizmo) [inline flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testutils/formats/SchemaUtils.java:23]: Can you add the javadoc?

**Review by alpinegizmo — APPROVED** [review]
It seems unfortunate that the contract here has to be independently implemented in several different places.

I'm wondering if the same problem could also exist in any of the recently externalized connectors?

**Comment by zentol** [comment]
> It seems unfortunate that the contract here has to be independently implemented in several different places.

Yes :( The are just used in too many different places in too many different ways.

> I'm wondering if the same problem could also exist in any of the recently externalized connectors?

Possibly; I will check them as well after the feature freeze.

**Comment by alpinegizmo** [comment]
> > It seems unfortunate that the contract here has to be independently implemented in several different places.
> 
> Yes :( The are just used in too many different places in too many different ways.

Yeah. I wanted to find something to abstract away, but gave up once I saw the diversity involved.

**Pull request merged**